### ISSUE TITLE
[Bug]: Can't update covers on ebooks or audiobooks

### ISSUE THREAD

**Issue by Spectre216** [advplyr/audiobookshelf#1901]
### Describe the issue

When I go to update covers via the web interface I either get an error that the socket is disconnected (and then reconnect a second later with no cover update) or the URL could not be downloaded.
![SCR-20230707-ilhw](https://github.com/advplyr/audiobookshelf/assets/22502482/45ef2c30-bc6c-4f35-b01c-4b5865d26bf8)
![SCR-20230707-ilma](https://github.com/advplyr/audiobookshelf/assets/22502482/8736106e-9ef5-4412-8431-fcd9146cbc3f)

### Steps to reproduce the issue

1.     Log in through web interface and try to update covers. Neither manual upload, URL Search, or manual URL are working.

### Audiobookshelf version

2.2.23 (Debian)

### How are you running audiobookshelf?

Debian/PPA

**Comment by advplyr** [comment]
I'm not able to reproduce this. Most likely the reason the socket is disconnecting is because the server is crashing.

You will have to check the logs on the server for this crash error.

`journalctl -u audiobookshelf`

**Comment by joem569** [comment]
I have the same issue, and it has to do with permissions. I'm still working on figuring out the solution for me though. I have my install setup to use a mounted SMB share to store my audiobook library and what I found is that i get this error when trying to save the book cover to that share. If I change the setting in Audiobookshelf to not store the cover with the item, then it works. This isn't ideal for me, I'd really like to keep all the files together.

I got around this initially by manually saving my cover pictures and putting them into the folders with the audio files, naming the cover file "cover.jpg" and this worked great. But I'm making another attempt to get is to work through Audiobookshelf and I keep hitting this issue. It has to be a permissions thing between my Audiobookshelf Docker container and how I have my SMB share mounted. I'm not a Docker guru, but that's what I think so far.

PS, I also encountered the same kinds of issues when I tried to save the metadata with the item too, but I set that to off and I haven't gotten that error since.
